src/features/assets/utils/arc62.ts

```diff
@@ -0,0 +1,101 @@
+import { indexer } from '@/features/common/data/algo-client'
+import algosdk, { getApplicationAddress } from 'algosdk'
+import { Arc3MetadataResult, Arc62MetadataResult } from '../data/types'
+import { uint8ArrayToUtf8 } from '@/utils/uint8-array-to-utf8'
+import { executeFundedDiscoveryApplicationCall } from '@/utils/funded-discovery'
+
+// Checks if the asset metadata has the ARC-62 property in the correct place
+// Returns the ARC-62 application ID if present, otherwise undefined
+export const getArc62AppId = (asset: Arc3MetadataResult): bigint | undefined => {
+  const arc62 = asset?.metadata?.properties?.['arc-62']
+  if (!arc62 || typeof arc62 !== 'object' || Array.isArray(arc62)) return undefined
+
+  const appId = (arc62 as Record<string, unknown>)['application-id']
+  if (typeof appId === 'number' && Number.isInteger(appId) && appId >= 0) {
+    return BigInt(appId)
+  }
+  if (typeof appId === 'string' && /^\d+$/.test(appId)) {
+    return BigInt(appId)
+  }
+
+  return undefined
+}
+
+// TODO Arthur - Refactor this - Try to make funded account application discovery more reusable
+export const getArc62CirculatingSupply = async (applicationId: bigint, assetId: bigint) => {
+  const arc62GetCirculatingSupplyMethod = algosdk.ABIMethod.fromSignature('arc62_get_circulating_supply(uint64)uint64')
+
+  const arc62ContractData = await indexer.lookupApplications(applicationId).do()
+
+  if (!arc62ContractData.application) {
+    throw new Error(`Application with ID ${applicationId} not found.`)
+  }
+
+  const globalAppState = arc62ContractData.application?.params.globalState ?? []
+  const decodedState = decodeAppState(globalAppState)
+  const burnedAddress = decodedState.burned.toString()
+
+  if (!burnedAddress) {
+    throw new Error(`Burned address not found in application ${applicationId} global state.`)
+  }
+
+  const simulateResult = await executeFundedDiscoveryApplicationCall(arc62GetCirculatingSupplyMethod, applicationId, [assetId])
+
+  if (!simulateResult || !simulateResult.returns?.[0]) return
+  return simulateResult.returns[0].returnValue
+}
+
+export const getArc62BurnedSupply = async (applicationId: bigint, assetId: bigint) => {
```

Comment:
These functions to get the specific supply amounts probably shouldn't ever throw, so that it acts like a progressive enhancement where available. If we can't determine the amount simply return undefined.